neuraxle/steps/util.py

```diff
@@ -316,12 +318,89 @@ class DataShuffler:
 
 
 class NullPipelineSaver(PipelineSaver):
+    def load(self, pipeline: 'Pipeline', data_container: DataContainer) -> 'Pipeline':
+        return pipeline
+
     def can_load(self, pipeline: 'Pipeline', data_container: DataContainer) -> bool:
         return True
 
     def save(self, pipeline: 'Pipeline', data_container: DataContainer) -> 'Pipeline':
         return pipeline
 
+
+class ForEachDataInputs(TruncableSteps):
+    """
+    Truncable step that fits/transforms each step for each of the data inputs, and expected outputs.
+    """
+
+    def fit(self, data_inputs, expected_outputs=None):
+        """
+        Fit each step for each data inputs, and expected outputs
+
+        :param data_inputs:
+        :param expected_outputs:
+        :return: self
+        """
+        if expected_outputs is None:
+            expected_outputs = [None] * len(data_inputs)
+
+        new_steps_as_tuple: NamedTupleList = []
+
+        for name, step in self.items():
+            for di, eo in zip(data_inputs, expected_outputs):
+                step = step.fit(di, eo)
+                new_steps_as_tuple.append(step)
+            data_inputs = data_inputs
```

Comment:
There seems to be three bugs here: 
1. You forgot to save the already-existing name back into the new steps as tuple. We don't want to refresh to new names because we don't change the nature of the existing objects. 
2. It seems to me that the line `new_steps_as_tuple.append(step)` should be unindented and be done in the first for loop, not in the second, and that this line should add the existing name as I stated in point 1 above. 
3. But anyways, fixing the point 1. and 2. isn't important, because we'll change the TruncableSteps to a MetaStepsMixin, so we won't have to ever loop on self.items() and a loop will disappear here.